config/routes.rb

```diff
@@ -304,10 +297,91 @@
+  get "u/:username/preferences/username" => "users#preferences", constraints: {username: USERNAME_ROUTE_FORMAT}
+  put "u/:username/preferences/username" => "users#username", constraints: {username: USERNAME_ROUTE_FORMAT}
+  delete "u/:username/preferences/user_image" => "users#destroy_user_image", constraints: {username: USERNAME_ROUTE_FORMAT}
+  put "u/:username/preferences/avatar/pick" => "users#pick_avatar", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/preferences/card-badge" => "users#card_badge", constraints: {username: USERNAME_ROUTE_FORMAT}
+  put "u/:username/preferences/card-badge" => "users#update_card_badge", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/staff-info" => "users#staff_info", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/summary" => "users#summary", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/invited" => "users#invited", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/invited_count" => "users#invited_count", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/invited/:filter" => "users#invited", constraints: {username: USERNAME_ROUTE_FORMAT}
+  post "u/action/send_activation_email" => "users#send_activation_email"
+  get "u/:username/summary" => "users#show", constraints: {username: USERNAME_ROUTE_FORMAT}
+
+  # user activity RSS feed
+  get "u/:username/activity/topics.rss" => "list#user_topics_feed", format: :rss, constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/activity.rss" => "posts#user_posts_feed", format: :rss, constraints: {username: USERNAME_ROUTE_FORMAT}
+
+  get "u/:username/activity" => "users#show", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/activity/:filter" => "users#show", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/badges" => "users#show", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/notifications" => "users#show", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/notifications/:filter" => "users#show", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/activity/pending" => "users#show", constraints: {username: USERNAME_ROUTE_FORMAT}
+  delete "u/:username" => "users#destroy", constraints: {username: USERNAME_ROUTE_FORMAT}
+  # The external_id constraint is to allow periods to be used in the value without becoming part of the format. ie: foo.bar.json
+  get "u/by-external/:external_id" => "users#show", constraints: {external_id: /[^\/]+/}
+  get "u/:username/flagged-posts" => "users#show", constraints: {username: USERNAME_ROUTE_FORMAT}
+  get "u/:username/deleted-posts" => "users#show", constraints: {username: USERNAME_ROUTE_FORMAT}
+
+  get "u/:username/topic-tracking-state" => "users#topic_tracking_state", constraints: {username: USERNAME_ROUTE_FORMAT}
+
+  # Old User routes, will be removed when /u/ is everywhere
```

Comment:
Ah ok :+1:  I guess we'll have to deprecate `users/` routes and make it known to client libraries somehow?